### PR TITLE
Serial: Allow for distinct execution space instances

### PR DESCRIPTION
**Pull request by masterleinad**
We discussed that we want to allow for distinct `Serial` execution space instances before merging https://github.com/kokkos/kokkos/pull/6151.
This pull request introduces a `NewInstance` tag that gives `Serial` instances that are different from the default constructed one.

Drive-by: Remove a test case that should never be executed because we assert before that `sum1==sum2`.

### CODE REVIEW

File: core/src/Serial/Kokkos_Serial.hpp [hunk 1]
```diff
@@ -72,6 +72,8 @@ class SerialInternal {
 };
 }  // namespace Impl
 
+struct NewInstance {};
```
Comment (nliber) [inline core/src/Serial/Kokkos_Serial.hpp:75]: This should be

```c++
struct NewInstance { explicit NewInstance() = default; };
```

to disallow calling it via:

```c++
Serial({});
```

**Review by nliber — DISMISSED** [review]

File: core/src/Serial/Kokkos_Serial.hpp (hunk 1 above)
Comment (masterleinad) [reply core/src/Serial/Kokkos_Serial.hpp]: 297e7e3

**Review by dalg24 — APPROVED** [review]

**Review by ldh4 — APPROVED** [review]

**Comment by Rombur** [comment]
clang format is not happy

**Comment by masterleinad** [comment]
> clang format is not happy

Fixed.

**Comment by masterleinad** [comment]
Only the `GCC-8.4.0` CI is timing out. Everything else passed.

**Review by nliber — APPROVED** [review]

**Pull request merged**